pkg/autodiscovery/listeners/kubelet.go

```diff
@@ -47,7 +47,7 @@ func NewKubeletListener(Config) (ServiceListener, error) {
 	l := &KubeletListener{}
 	f := workloadmeta.NewFilter(
 		[]workloadmeta.Kind{workloadmeta.KindKubernetesPod},
-		[]workloadmeta.Source{workloadmeta.SourceKubelet},
+		workloadmeta.SourceNodeOrchestrator,
```

Comment:
It won't receive ECS events because there's another filter for `KindKubernetesPod`, so in that sense it's still very much tied to k8s and is not portable to other orchestrators. It could be renamed into a "pod" listener or something, but I don't think it's worth the effort at this point.